org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/lifecyclemapping/DuplicateMappingException.java

```diff
@@ -13,6 +13,36 @@
 
 package org.eclipse.m2e.core.internal.lifecyclemapping;
 
+import org.eclipse.m2e.core.internal.lifecyclemapping.model.LifecycleMappingMetadata;
+import org.eclipse.m2e.core.internal.lifecyclemapping.model.LifecycleMappingMetadataSource;
+import org.eclipse.m2e.core.internal.lifecyclemapping.model.PluginExecutionMetadata;
+
+
+/**
+ * Exception thrown when multiple metadata source for either {@link PluginExecutionMetadata} or
+ * {@link LifecycleMappingMetadata} are found for the same plugin/packaging.
+ */
 public class DuplicateMappingException extends RuntimeException {
+  private static final String DESCRIPTION_UNKNOWN_SOURCE = "unknown source";
+
   private static final long serialVersionUID = 6916144930019743563L;
+
+  private final LifecycleMappingMetadataSource source1, source2;
```

Comment:
One line per field would be a bit better style.

Reply:
https://github.com/eclipse-m2e/m2e-core/pull/530/commits/78530ec488ad407700490a48b6bed5709e1ee3e3